### PR TITLE
Fix lookup of solv-file path for repos with '/' in name (bsc#1037101)

### PR DESCRIPTION
**Pull request by okurz**
zypp auto-replaces '/' in names with '_' so we need to do the same here to
find the correct solv-files.

### CODE REVIEW

File: scout/bin.py [hunk 1]
```diff
@@ -23,20 +24,29 @@ class SolvParser(object):
...
+        zypper_refresh_failed = False
         for repofile in [ f for f in os.listdir(self.etcpath) if fnmatch(f, '*.repo') ]:
             try:
                 parser = SafeConfigParser()
                 parser.read( '%s/%s' % (self.etcpath, repofile) )
                 for name in parser.sections():
                     if parser.get(name, 'enabled') == '1':
-                        if not os.path.isfile(self.solvfile % name):
-                            os.system('zypper refresh')
+                        if not os.path.isfile(self.solvfile % name.replace('/', '_')) and not zypper_refresh_failed:
```
Comment (aplanas) [inline scout/bin.py]: Still is not python3, we need to wait a bit.

File: scout/bin.py [hunk 2]
```diff
@@ -23,20 +24,29 @@ class SolvParser(object):
...
+                            ret = call('zypper refresh')
+                            assert ret, "zypper refresh failed with exit code: %s" % ret
                         repo = self.pool.add_repo(name)
                         repo.add_solv(self.solvfile % name)
+            except OSError as e:
+                import pdb
```
Comment (aplanas) [inline scout/bin.py]: ups : )

File: scout/bin.py (hunk 1 above)
Comment (okurz) [reply scout/bin.py]: Uhm, I tested with python 2.7 . did I do something wrong? Anyway, I guess I could test with a venv of older python

Comment (aplanas) [reply scout/bin.py]: Oh I see the issue. Is `isfile`:

```
Python 2.7.14 (default, Oct 12 2017, 15:50:02) [GCC] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os.path
>>> os.path.is_file
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'is_file'
>>> os.path.isfile
<function isfile at 0x7fcd1bf76f50>
>>> 
```

**Comment by okurz** [comment]
done

**Pull request merged**